### PR TITLE
react todo app solution

### PR DESCRIPTION
**Pull request by opiquem**
[DEMO LINK](https://maksym-bobryk.github.io/react_todo-app-with-api/)

### CODE REVIEW

**Review by vladyslav-lapshyn — APPROVED** [review]
As for me it looks amazing 🔥🔥🔥
I just left some sugestions. GOOD JOB 🔥

File: src/components/TodoInfo/TodoInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      handleSubmitTitle();
+    }
+
+    if (event.key === 'Escape') {
+      event.preventDefault();
+      handleCancelTitle();
+    }
+  };
```
Comment (33Orange) [inline src/components/TodoInfo/TodoInfo.tsx:60]:
```suggestion
  const handleKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
    event.preventDefault();
    
    if (event.key === 'Escape') {
      handleCancelTitle();
      return;
    }

     handleSubmitTitle();
  };
```

File: src/components/TodoInfo/TodoInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            className="todo__title"
+            onDoubleClick={handleEditTitle}
+          >
+            {todo.title}
+          </span>
+
+          <button
+            type="button"
+            className="todo__remove"
+            onClick={() => onRemoveTodo(todo.id)}
```
Comment (33Orange) [inline src/components/TodoInfo/TodoInfo.tsx:112]: Create handler function

File: src/components/TodoInfo/TodoInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+          >
+            ×
+          </button>
+        </>
+      )}
+
+      <div className={cn('modal overlay', {
+        'is-active': isLoadingTodo.includes(todo.id),
```
Comment (33Orange) [inline src/components/TodoInfo/TodoInfo.tsx:120]: Create variable for this flag

**Review by 33Orange — CHANGES_REQUESTED** [review]
Almost done

A few moments to fix and improve

File: src/components/TodoFooter/TodoFooter.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+// import { Todo } from '../../types/Todo';
+
+type Props = {
+  status: TodoStatusFilter,
+  onSelectStatusFilter: (status: TodoStatusFilter) => void,
+  uncompletedTodosCount: number,
+  isVisibleClearCompleted: boolean,
+  onRemoveAllCompletedTodos: () => void,
+  // todos: Todo[],
+  // onRemoveTodo: (todoId: number) => void,
```
Comment (33Orange) [inline src/components/TodoFooter/TodoFooter.tsx:13]: Remove redundant comments

File: src/components/TodoFooter/TodoFooter.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { TodoFilter } from '../TodoFilter/TodoFilter';
+import { TodoStatusFilter } from '../../types/TodoStatusFilter';
+// import { Todo } from '../../types/Todo';
```
Comment (33Orange) [inline src/components/TodoFooter/TodoFooter.tsx:4]: Same

**Review by 33Orange — APPROVED** [review]
Looks good. Good job

Remove redundant comments

**Pull request closed without merging**